src/transformers/models/vit/feature_extraction_vit.py

```diff
@@ -0,0 +1,136 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Feature extractor class for ViT."""
+
+from typing import List, Union
+
+import numpy as np
+import torch
+from PIL import Image
+from torchvision import transforms as T
+
+from ...feature_extraction_utils import BatchFeature, FeatureExtractionMixin
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+class ViTFeatureExtractor(FeatureExtractionMixin):
+    r"""
+    Constructs a ViT feature extractor.
+
+    This feature extractor inherits from :class:`~transformers.FeatureExtractionMixin` which contains most of the main
+    methods. Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+        image_mean (:obj:`int`, defaults to [0.5, 0.5, 0.5]):
+            The sequence of means for each channel, to be used when normalizing images.
+        image_std (:obj:`int`, defaults to [0.5, 0.5, 0.5]):
+            The sequence of standard deviations for each channel, to be used when normalizing images.
+        do_normalize (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Whether or not to normalize the input with mean and standard deviation.
+        do_resize (:obj:`bool`, `optional`, defaults to :obj:`True`):
+            Whether to resize the input to a certain :obj:`size`.
+        size (:obj:`int`, `optional`, defaults to :obj:`224`):
+            Resize the input to the given size. Only has an effect if :obj:`do_resize` is set to :obj:`True`.
+    """
+
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self,
+        image_mean=[0.5, 0.5, 0.5],
+        image_std=[0.5, 0.5, 0.5],
+        do_normalize=True,
+        do_resize=True,
+        size=224,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.image_mean = image_mean
+        self.image_std = image_std
+        self.do_normalize = do_normalize
+        self.do_resize = do_resize
+        self.size = size
+
+    def __call__(
+        self,
+        images: Union[Image.Image, np.ndarray, torch.Tensor, List[Image.Image], List[np.ndarray], List[torch.Tensor]],
+        **kwargs
+    ) -> BatchFeature:
+        """
+        Main method to prepare for the model one or several image(s).
+
+        Args:
+            images (:obj:`PIL.Image.Image`, :obj:`np.ndarray`, :obj:`torch.Tensor`, :obj:`List[PIL.Image.Image]`, :obj:`List[np.ndarray]`, :obj:`List[torch.Tensor]`):
+                The image or batch of images to be prepared. Each image can be a PIL image, numpy array or a Torch
+                tensor. In case of a numpy array/Torch tensor, each image should be of shape (C, H, W), where C is a
```

Comment:
```suggestion
                The image or batch of images to be prepared. Each image can be a PIL image, numpy array or a PyTorch
                tensor. In case of a NumPy array/PyTorch tensor, each image should be of shape (C, H, W), where C is a
```